platform/linux-dpdk/Makefile.am

```diff
@@ -297,6 +300,17 @@ __LIB__libodp_dpdk_la_SOURCES += arch/x86/cpu_flags.c \
 endif
 
 pool/dpdk.lo: AM_CFLAGS += -DIM_ACTIVE_MODULE
+if ODP_SCHEDULE_SCALABLE
```

Comment:
My understanding is that we are building and registering all scheduler variants and here only set the active one.
Is my understanding OK? 
For ODP_SCHEDULE_SP case, we will have both 'generic' and 'sp' built with IM_ACTIVE_MODULE. Is this OK from modular framework point of view?

Reply:
Yes, only one scheduler is marked as the 'active' scheduler. This is controlled by ```--enable-schedule-xxx``` configure flag.

@heyi-linaro is working on build system improvements in this area, so please expect this to evolve.

Reply:
@brbrooks actually when using  --enable-schedule-sp you will have both 'generic' and 'sp' marked as active.
I was thinking at:

if ODP_SCHEDULE_SCALABLE
schedule/scalable.lo: AM_CFLAGS += -DIM_ACTIVE_MODULE
else
if ODP_SCHEDULE_SP
schedule/sp.lo: AM_CFLAGS += -DIM_ACTIVE_MODULE
else
if ODP_SCHEDULE_IQUERY
schedule/iquery.lo: AM_CFLAGS += -DIM_ACTIVE_MODULE
else
schedule/generic.lo: AM_CFLAGS += -DIM_ACTIVE_MODULE
endif
endif
endif

Is ugly, but does the job. 
Yet, your code may work... this was more a question for @heyi-linaro .

Reply:
Thanks @bogdanPricope. It would be nice to have ```elif``` or ```else if```. :relaxed:

Reply:
True, but there is no 'elif' / 'else if' construction available from what I know....

Reply:
Yes, in next plan these "ugly" conditional rules should be auto-generated, subsystem and module writers only need to "declare" their subsystem and modules nicely and concisely like:

SUBSYSTEMS += new_subsystem
__new_subsystem_MODULES += one:default \
                                                      two:SELECT_FLAG_A \
                                                      three: SELECT_FLAG_B ...